### PR TITLE
fix(ESSNTL-5239): Improve RBAC check for bulk group actions

### PR DESCRIPTION
**Pull request by gkarat**
Implements https://issues.redhat.com/browse/ESSNTL-5239.

Must be merged after https://github.com/RedHatInsights/insights-inventory-frontend/pull/1973 and https://github.com/RedHatInsights/frontend-components/pull/1885.

This makes sure that both **bulk** actions (Remove from group and Add to group) on the /inventory page comply with the RBAC requirements: specifically that the table no longer checks on the global level permissions.

- [x] Write cy tests

## How to test

1. Set up `inventory:groups:write` permission only to specific group
2. Navigate to /inventory
3. Select one or more ungroupped hosts and make sure "Add to group" is enabled; try to add the groups 
4. Select some systems in a group that you don't have write permissions: you shouldn't be able to bulk remove them from the group
5. Select some systems in a group that you have write permissions: you should be able to remove them via bulk Remove from group 
6. Select the same systems as in 5. + some ungroupped hosts: you still should be able to remove the hosts that are in the group, and the modal should correctly display the number of hosts to be removed.

### CODE REVIEW

File: src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js [hunk 1]
```diff
@@ -7,62 +7,78 @@ import { removeHostsFromGroup } from '../utils/api';
...
 const RemoveHostsFromGroupModal = ({
   isModalOpen,
   setIsModalOpen,
   modalState: hosts,
   reloadData,
 }) => {
-  const dispatch = useDispatch();
-  // the current iteration of groups feature a host can be in at maximum one group
-  const { name: groupName, id: groupId } = hosts[0].groups[0];
+  // TODO: should soon support removal from more than one group at once
```
Comment (LiorKGOW) [inline src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js:67]: Do you intend to keep this comment ?
If not, this is just a reminder to remove it :)

File: src/routes/InventoryTable.cy.js [hunk 2]
```diff
@@ -339,8 +340,104 @@ describe('inventory table', () => {
...
+        cy.get(ROW).eq(4).find(DROPDOWN).click();
+        cy.get(DROPDOWN_ITEM).contains('Add to group').shouldHaveAriaEnabled();
+        cy.get(DROPDOWN_ITEM)
+          .contains('Remove from group')
+          .shouldHaveAriaDisabled();
+      });
+
+      it('can bulk remove from the permitted group', () => {
+        cy.get(ROW).find('[type="checkbox"]').eq(0).click();
+        // TODO: implement ouia selector for this component
```
Comment (LiorKGOW) [inline src/routes/InventoryTable.cy.js:393]: here

File: src/routes/InventoryTable.cy.js [hunk 3]
```diff
@@ -339,8 +340,104 @@ describe('inventory table', () => {
...
+        cy.get(
+          '.ins-c-primary-toolbar__actions [aria-label="Actions"]'
+        ).click();
+        cy.get(DROPDOWN_ITEM)
+          .contains('Remove from group')
+          .shouldHaveAriaEnabled();
+        cy.get(ROW).find('[type="checkbox"]').eq(1).click();
+        // TODO: implement ouia selector for this component
```
Comment (LiorKGOW) [inline src/routes/InventoryTable.cy.js:401]: here

File: src/routes/InventoryTable.cy.js [hunk 4]
```diff
@@ -339,8 +340,104 @@ describe('inventory table', () => {
...
+        ).click();
+        cy.get(DROPDOWN_ITEM)
+          .contains('Remove from group')
+          .shouldHaveAriaEnabled();
+      });
+
+      it('can bulk remove from group together with ungroupped hosts', () => {
+        cy.get(ROW).find('[type="checkbox"]').eq(0).click();
+        cy.get(ROW).find('[type="checkbox"]').eq(3).click();
+        // TODO: implement ouia selector for this component
```
Comment (LiorKGOW) [inline src/routes/InventoryTable.cy.js:413]: here

File: src/routes/InventoryTable.cy.js [hunk 5]
```diff
@@ -339,8 +340,104 @@ describe('inventory table', () => {
...
+        ).click();
+        cy.get(DROPDOWN_ITEM)
+          .contains('Remove from group')
+          .shouldHaveAriaEnabled();
+      });
+
+      it('can bulk add hosts to the permitted group', () => {
+        cy.get(ROW).find('[type="checkbox"]').eq(3).click();
+        cy.get(ROW).find('[type="checkbox"]').eq(4).click();
+        // TODO: implement ouia selector for this component
```
Comment (LiorKGOW) [inline src/routes/InventoryTable.cy.js:425]: here

File: src/routes/InventoryTable.cy.js [hunk 6]
```diff
@@ -339,8 +340,104 @@ describe('inventory table', () => {
...
+        cy.get(
+          '.ins-c-primary-toolbar__actions [aria-label="Actions"]'
+        ).click();
+        cy.get(DROPDOWN_ITEM).contains('Add to group').shouldHaveAriaEnabled();
+      });
+
+      it('cannot bulk add to group if groupped hosts selected', () => {
+        cy.get(ROW).find('[type="checkbox"]').eq(0).click();
+        cy.get(ROW).find('[type="checkbox"]').eq(3).click();
+        // TODO: implement ouia selector for this component
```
Comment (LiorKGOW) [inline src/routes/InventoryTable.cy.js:435]: here

File: src/routes/InventoryTable.js [hunk 7]
```diff
@@ -270,24 +272,23 @@ const Inventory = ({
...
+      ).length === 1
+    );
   };
 
   const isBulkAddHostsToGroupsEnabled = () => {
     if (calculateSelected() > 0) {
       const selectedHosts = Array.from(selected.values());
 
-      return selectedHosts.every(({ groups }) => groups.length === 0);
+      return selectedHosts.every(({ groups }) => groups.length === 0); // TODO: to be removed soon
```
Comment (LiorKGOW) [inline src/routes/InventoryTable.js:291]: here

File: src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js [hunk 8]
```diff
@@ -7,62 +7,78 @@ import { removeHostsFromGroup } from '../utils/api';
...
+      },
+    ],
+  };
+};
+
+const statusMessages = (hosts) => {
+  const hostsInGroup = hosts.filter(({ groups }) => groups.length > 0);
+  const groupName = hostsInGroup[0].groups[0].name;
+
+  return hostsInGroup.length === 1
```
Comment (LiorKGOW) [inline src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js:42]: You could also put all the checks inside the string, using a Template Literal with the ternary operator, and by that remove some code duplication :)

Something like this:
```
const numOfSystems = hostsInGroup.length;
return {
onSuccess: {
title: `${numOfSystems} system${numOfSystems===1 ? " was" : "s were"} removed from ${groupName}`,
},
onError: {
title: `Failed to remove ${numOfSystems} system${numOfSystems===1 ? "" : "s"} from ${groupName}`,
},
};
```

File: src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js [hunk 9]
```diff
@@ -7,62 +7,78 @@ import { removeHostsFromGroup } from '../utils/api';
...
+  const hostsInGroup = hosts.filter(({ groups }) => groups.length > 0); // selection can contain ungroupped hosts
+  const groupName = hostsInGroup[0].groups[0].name;
+
+  return {
+    fields: [
+      {
+        component: componentTypes.PLAIN_TEXT,
+        name: 'warning-message',
+        label:
+          hostsInGroup.length === 1 ? (
```
Comment (LiorKGOW) [inline src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js:20]: These lines can also be shortened using the template literal with the ternary operator 👍🏼 
Something like this:
```
const numOfSystems = hostsInGroup.length;
const systemName = hostsInGroup[0].display_name;
label:
  <Text>
    <strong>{numOfSystems === 1 ? systemName : numOfSystems}</strong>{numOfSystems === 1 ? "" : " 
    systems"} will no longer be part of <strong>{groupName}</strong> and {numOfSystems === 1 ? "its" : 
    "their"} configuration will be impacted.
  </Text>
```

**Review by LiorKGOW — COMMENTED** [review]
Nice work overall @gkarat 
I have found a few comments which I am not sure if you intended to keep (marked them with 'here')

Also, I think you could be using the template literal alongside with the ternary operator to shorten the code and remove some code duplication :)
I have added an example of how I thought it could go
In the file: src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js
1. line 42
2. line 20

File: src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js (hunk 1 above)
Comment (gkarat) [reply src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js:67]: Yeah, it doesn't have any special call for action, but I rather wanted to keep it as helping tip. But I better remove it completely

File: src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js (hunk 8 above)
Comment (gkarat) [reply src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js:42]: Yeah this is the variant that used to be here. But I decided to rework it for easier code readability (even though it's not ideally following DRY).

File: src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js (hunk 9 above)
Comment (gkarat) [reply src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js:20]: Thank you. But still, I would like to stick to the current variant (see https://github.com/RedHatInsights/insights-inventory-frontend/pull/1974#discussion_r1294420391 comment)

**Pull request merged**